### DISCUSSION TITLE
Allow shorter artifact retention times

### DISCUSSION THREAD

**Discussion by aramfe** [community/community#12028] [Actions]
Right now one set up the retention time for artifact deletion for anything lower than minimum one day. That's unfortunate.

We would also like to have the option to delete them in smaller retention times, like six or twelve hours. We also don't like, that we can't separate logs and artifacts in terms of their retention times, as we would like to have longer lasting logs, but shorter lasting artifacts. 

We had to implement our own Cron-Job as it's own workflow, that deletes artifacts every six hours, just because there isn't any other option to have shorter retention times than one day.

**Comment by ethomson** [comment]
Thanks for the feedback @aramfe.  Can you tell me more about why you want a tighter retention?  Are you just producing a lot of data and want to keep the amount used under control, or are there some regulatory concerns that you have?

**Reply by aramfe** [reply]
In fact for both cases: It unnecessarily takes space of the maximum storage included in our enterprise plan and could also lead potentially to bigger problems regarding regulatory concerns, as we are based in a very strict european country in terms of data security. 

So keeping their retention time as tight as possible is a must for us.

**Reply by ethomson** [reply]
Thanks - I appreciate the additional details.

**Reply by michael-otmar-kaiser-db** [reply]
I also would like to keep the workflow summary page clean. We use `upload-artifact` in combination with `download-artifact` pretty often just to pass artifacts from one job to another. We don't need the intermediate job results anymore once the workflow completed. I would love to just set the retention time to just a few seconds or minutes.

**Reply by kolpav** [reply]
@ethomson Hi, if  you need more details you can take a look at my https://github.com/actions/upload-artifact/issues/290 aggregating multiple issues from people having the same problem. I would love to get **any** feedback on it even knowing its not going to be fixed or acknowledgement that what I and many other people are experiencing is indeed a problem.
